detectable/src/main/java/com/blackduck/integration/detectable/detectables/gradle/inspection/GradleInspectorExtractor.java

```diff
@@ -66,12 +67,23 @@ public Extraction extract(File directory, ExecutableTarget gradleExe, @Nullable
             reportFiles.toArray(files);
             List<File> reportFilesSorted = Arrays.asList(sortFilesByDepth(files));
 
-            reportFilesSorted.stream()
-                .map(gradleReportParser::parseReport)
-                .filter(Optional::isPresent)
-                .map(Optional::get)
-                .map(gradleReportTransformer::transform)
-                .forEach(codeLocations::add);
+            if (rootOnly) {
+                logger.debug("Gradle Inspector root only option selected. Will process root project's dependencies only.");
+                Optional<GradleReport> rootReport = gradleReportParser.parseReport(reportFilesSorted.get(0));
+                if (!rootReport.isPresent()) {
+                    logger.error("No root dependency report to process."); // TODO fail gradle inspector?
```

Comment:
Testing to see how to fail here so the detector report is accurate (_"Gradle Native Inspector attamepted and failed"_), please ignore for now.